eng/pipelines/templates/pipeline.yml

```diff
@@ -64,6 +64,17 @@ stages:
     steps:
     - template: Initialize_Build.yml
 
+- stage: Scan_Code
+  displayName: Pre-Checks before start build
```

Comment:
This isn't running before the build starts, so the display name isn't right. But I want to move the `dotnet-format` check into this pipeline (meaning Scan_Code probably also isn't a good name). I know names are hard, but just an FYI to think about in future PRs to have the highest quality PRs for the team to review.